xbmc/windowing/wayland/WinSystemWayland.cpp

```diff
@@ -592,11 +592,21 @@ bool CWinSystemWayland::SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, boo
       ApplyBufferScale(m_scale);
     }
 
-    // FIXME This assumes that the resolution has already been set. Should
-    // be moved to some post-change callback when resolution setting is refactored.
-    if (!m_inhibitSkinReload)
+    if (m_shellSurfaceState.test(IShellSurface::STATE_RESIZING) && !m_resizeSkinReloadTimer.IsRunning())
     {
-      g_application.ReloadSkin();
+      // Reloading the skin is costly -> during resizing (when the size changes quite
```

Comment:
That would be really great since reloading is quite slow. How can we fix the aspect ratio? It's not limited to windowed mode by the way, it also affects full screen.